library/ssl_tls13_client.c

```diff
@@ -2505,6 +2541,22 @@ static int ssl_tls13_parse_new_session_ticket_exts( mbedtls_ssl_context *ssl,
 
         switch( extension_type )
         {
+            case MBEDTLS_TLS_EXT_EARLY_DATA:
+                MBEDTLS_SSL_DEBUG_MSG( 4, ( "early_data extension received" ) );
+                if( extension_data_len != 4 )
+                {
+                    MBEDTLS_SSL_PEND_FATAL_ALERT(
+                        MBEDTLS_SSL_ALERT_MSG_DECODE_ERROR,
+                        MBEDTLS_ERR_SSL_DECODE_ERROR );
+                    return( MBEDTLS_ERR_SSL_DECODE_ERROR );
+                }
+                if( ssl->session != NULL )
+                {
+                    ssl->session->ticket_flags |=
+                            MBEDTLS_SSL_TLS1_3_TICKET_ALLOW_EARLY_DATA;
```

Comment:
One more comment. I did not found test case for having early_data or not having early_data.

I am not sure how to do that, but we can mark a todo item in this PR.

Reply:
If we have early data, we will send out the early data extension in the client hello, or will not send out it.  Does that mean we need one negative test cases without early data In new session tickets?

Reply:
That need a server does not send early data and a server send early data.

Reply:
I just add one negative case named: **TLS 1.3 m->G: EarlyData: negative check, fail**, the server will not send out the early data, please help see whether it can satisfy the requirement.